src/realm/object-store/sync/app.cpp

```diff
@@ -257,26 +257,39 @@ void App::configure(const SyncClientConfig& sync_client_config)
 {
     auto sync_route = make_sync_route(m_app_route);
     m_sync_manager->configure(shared_from_this(), sync_route, sync_client_config);
-    m_logger = m_sync_manager->make_logger();
     if (auto metadata = m_sync_manager->app_metadata()) {
         update_hostname(metadata);
     }
 }
 
+void App::set_logger(const std::shared_ptr<util::Logger>& logger) noexcept
+{
+    std::lock_guard<std::mutex> lock(*m_logger_mutex);
+    m_logger_ptr = logger;
+}
+
+bool App::would_log(util::Logger::Level level)
+{
+    std::lock_guard<std::mutex> lock(*m_logger_mutex);
+    return m_logger_ptr && m_logger_ptr->would_log(level);
```

Comment:
At this point determining whether we would log is going to involve locking a mutex to know if we should format a few strings in non-hot paths? Having this function doesn't seem like it's buying us much performance-wise.

Reply:
This was added since there is a situation where `App` could potentially have `set_logger()` called twice:
 - one when the App is configured and the logger is created in `SyncManager::configure()`
 - and a second when a new logger is created when `SyncManager::SetLoggerFactory()` is called
I had actually broken the `SetLoggerFactory()` operation when I added logging to App in a previous commit. The App was creating/getting the logger when `App::Configure()` was called and it would not be recreated when `SetLoggerFactory()` was called.
Several of the SDKs use/expose the `SetLoggerFactory()` function, so I couldn't get rid of it...